src/plugins/controls/public/types.ts

```diff
@@ -56,51 +23,16 @@ export const isClearableControl = (control: unknown): control is CanClearSelecti
   return typeof (control as CanClearSelections).clearSelections === 'function';
 };
 
-/**
- * Control embeddable editor types
- */
-export interface IEditableControlFactory<T extends ControlInput = ControlInput>
-  extends Pick<EmbeddableFactory, 'type'> {
-  controlEditorOptionsComponent?: (props: ControlEditorProps<T>) => JSX.Element;
-  presaveTransformFunction?: (
-    newState: Partial<T>,
-    embeddable?: ControlEmbeddable<T>
-  ) => Partial<T>;
-  isFieldCompatible?: (field: DataViewField) => boolean;
-}
-
-export interface ControlEditorProps<T extends ControlInput = ControlInput> {
-  initialInput?: Partial<T>;
-  fieldType: string;
-  onChange: (partial: Partial<T>) => void;
-  setControlEditorValid: (isValid: boolean) => void;
-}
-
-export interface DataControlField {
-  field: DataViewField;
-  compatibleControlTypes: string[];
-}
-
-export interface DataControlFieldRegistry {
-  [fieldName: string]: DataControlField;
-}
-
-export interface DataControlEditorChanges {
-  input: Partial<DataControlInput>;
-  width?: ControlWidth;
-  grow?: boolean;
-}
-
 /**
  * Plugin types
  */
 export interface ControlsPluginSetup {
-  registerControlType: ControlsServiceType['registerControlType'];
+  registerControlFactory: ControlsServiceType['registerControlFactory'];
 }
 
 export interface ControlsPluginStart {
   getControlFactory: ControlsServiceType['getControlFactory'];
-  getControlTypes: ControlsServiceType['getControlTypes'];
+  getAllControlTypes: ControlsServiceType['getAllControlTypes'];
```

Comment:
I decided to keep the `setup` and `start` contracts more-or-less the same, despite the fact that no one is using them - if we want to remove the ability for outside plugins to register controls, we can delete this in a follow up PR.